packages/rum-core/src/domain/foregroundContexts.spec.ts

```diff
@@ -190,6 +214,49 @@ describe('foreground context', () => {
         })
       })
     })
+
+    describe(`after reaching the maximum number of focus periods: ${MAX_NUMBER_OF_FOCUSED_TIME}`, () => {
+      beforeEach(() => {
```

Comment:
what is the need for a separate suite and a beforeEach since there is a single test in it?

Reply:
I like to extract the setup because it allow me to add more test in it leveraging the nested structure of jest but you seems to prefer less verbose tests. I will refactor that